offer-manager-domain/src/main/groovy/life/qbic/business/products/ProductItem.java

```diff
@@ -59,9 +84,29 @@ public ProductItem(OfferV2 offer, Product product, Double quantity) {
     this.offer = requireNonNull(offer, "Offer must not be null");
     this.product = requireNonNull(product, "Product must not be null");
     this.quantity = requireNonNull(quantity, "Quantity must not be null");
+    copy(product);
+    this.product = product;
     refreshProductItem();
   }
 
+  private void copy(Product product) {
+    this.productCategory = product.getCategory();
+    this.description = product.getDescription();
+    this.productName = product.getProductName();
+    this.internalUnitPrice = product.getInternalUnitPrice();
+    this.externalUnitPrice = product.getExternalUnitPrice();
+    this.unit = product.getUnit();
+    this.productReference = product.getProductId();
+    this.serviceProvider = product.getServiceProvider();
+  }
+
+  @PostLoad
+  private void triggerProductCopy() {
+    if (Objects.isNull(this.productReference)) {
+      copy(product);
+    }
```

Comment:
What happens here?
In case the product is null (when is this the case?); Load information from the product.